### ISSUE TITLE
Not fixed bug with leaking string colors

### ISSUE THREAD

**Issue by usernamehw** [Tyriar/vscode-theme-generator#48]
You probably know it:
```javascript
func[0];
`text ${func[0]} text`;
```
![screenshot 12](https://user-images.githubusercontent.com/9638156/40575876-cea0c346-60f5-11e8-9c98-dd790592025e.png)

**Comment by Tyriar** [comment]
`[]` and `{}` don't actually appear to have assigned colors, adding this looks like it fixes it `meta.brace.square, punctuation.definition.block`. There are a lot of other brackets at least in the TS grammar that need to be covered

**Comment by usernamehw** [comment]
It's a known bug. It affects more that punctuation. I can't find the issue, but the way to fix it is to assign foreground color of each theme 

like this:
https://github.com/Microsoft/vscode/blob/master/extensions/theme-monokai-dimmed/themes/dimmed-monokai-color-theme.json#L58

or that:
https://github.com/Microsoft/vscode/blob/master/extensions/theme-monokai-dimmed/themes/dimmed-monokai-color-theme.json#L390

**Comment by Tyriar** [comment]
Fixed, let me know if you find other similar cases

**Comment by usernamehw** [comment]
No, I mean assigning foreground of those scopes that I linked above with the same color as `editor.foreground` is the official way of fixing this bug.

But if you want to search for every character:

Dot: `` `text ${func.length} text` ``
Comma: `` `text ${[0, 1]} text` ``
Braces: `` `text ${() => 0} text` ``

Probably more...

**Comment by Tyriar** [comment]
You mean targeting `meta.template.expression.ts meta.embedded.line.ts` instead? Probably a better idea...

**Comment by usernamehw** [comment]
No, I mean targeting the same scopes as all of the embedded vscode themes:
1. `"meta.embedded",`
1. `"source.groovy.embedded"`
1. `"meta.template.expression"`
